pkg/model/profile.go

```diff
@@ -45,3 +46,18 @@ func SelectorFromProfileType(profileType *typesv1.ProfileType) *labels.Matcher {
 		Value: profileType.Name + ":" + profileType.SampleType + ":" + profileType.SampleUnit + ":" + profileType.PeriodType + ":" + profileType.PeriodUnit,
 	}
 }
+
+// SetProfileMetadata sets the metadata on the profile.
+func SetProfileMetadata(p *profile.Profile, ty *typesv1.ProfileType) {
+	p.SampleType = []*profile.ValueType{{Type: ty.SampleType, Unit: ty.SampleUnit}}
+	p.DefaultSampleType = ty.SampleType
+	p.PeriodType = &profile.ValueType{Type: ty.PeriodType, Unit: ty.PeriodUnit}
+	switch ty.Name {
+	case "process_cpu": // todo: this should support other types of cpu profiles
```

Comment:
Should we create an issue for that?

Reply:
I already did. https://github.com/grafana/phlare/issues/560